test/new-e2e/tests/agent-platform/common/test_client.go

```diff
@@ -281,3 +270,63 @@ func ReadJournalCtl(t *testing.T, client *TestClient, grepPattern string) string
 	}
 	return journalCtlOutput
 }
+
+// DockerTestClient is a helper to run commands on a docker container for tests
+type DockerTestClient struct {
+	host          *components.RemoteHost
+	containerName string
+}
+
+// NewDockerTestClient creates a client to help write tests that run on a docker container
+func NewDockerTestClient(host *components.RemoteHost, containerName string) *DockerTestClient {
+	return &DockerTestClient{
+		host:          host,
+		containerName: containerName,
+	}
+}
+
+// RunContainer starts the docker container in the background based on the given image reference
+func (c *DockerTestClient) RunContainer(image string) error {
+	// We run an infinite no-op to keep it alive
+	_, err := c.host.Execute(
+		fmt.Sprintf("docker run -d -e DD_HOSTNAME=docker-test --name '%s' '%s' tail -f /dev/null", c.containerName, image),
+	)
+	return err
+}
+
+// Cleanup force-removes the docker container associated to the client
+func (c *DockerTestClient) Cleanup() error {
+	_, err := c.host.Execute(fmt.Sprintf("docker rm -f '%s'", c.containerName))
+	return err
+}
+
+// Execute runs commands on a Docker remote host
+func (c *DockerTestClient) Execute(command string) (output string, err error) {
+	return c.host.Execute(
+		// Run command on container via docker exec and wrap with sh -c
+		// to provide a similar interface to remote host's execute
+		fmt.Sprintf("docker exec %s sh -c '%s'", c.containerName, command),
+	)
+}
+
+// ExecuteWithRetry execute the command with retry
+func (c *DockerTestClient) ExecuteWithRetry(cmd string) (output string, err error) {
+	return execWithRetry(c.Execute, cmd)
+}
+
+func execWithRetry(exec func(string) (string, error), cmd string) (string, error) {
+	var err error
+	var output string
+	maxTries := 5
+
+	for try := 0; try < maxTries; try++ {
+		output, err = exec(cmd)
+		if err == nil {
+			break
+		}
+		fmt.Printf("(attempt %d of %d) error while executing %q: %v\n", try+1, maxTries, cmd, err)
```

Comment:
We need to be careful to not print tokens here.

Reply:
Good point, I'll remove the display of the executed command.